src/back/spv/writer.rs

```diff
@@ -1349,11 +1358,31 @@ impl Writer {
                     }
                 };
 
-                let (result_type_id, result_lookup_ty) = if result_side_left {
-                    (left_result_type_id, left_lookup_ty)
+                let is_comparison = match op {
+                    crate::BinaryOperator::Equal
+                    | crate::BinaryOperator::NotEqual
+                    | crate::BinaryOperator::Less
+                    | crate::BinaryOperator::LessEqual
+                    | crate::BinaryOperator::Greater
+                    | crate::BinaryOperator::GreaterEqual => true,
+                    _ => false,
+                };
+
+                let (result_type_id, result_lookup_ty) = if is_comparison {
+                    let local_ty = LookupType::Local(LocalType::Scalar {
```

Comment:
This would only work for scalars. I think we want IR to also support vector comparisons, i.e.
```
const xx: vec2<bool> = vec2<f32>(1.0, 1.0) == vec2<f32>(0.0, 1.0);
```

Reply:
True, I think at least, never used an expression like that! Will add it to my todo to create an issue for it at a later time.